agent/acs/handler/task_manifest_responder.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handler
+
+import (
+	"strconv"
+
+	apitaskstatus "github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/agent/data"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+// taskComparer implements the TaskComparer interface defined in ecs-agent module.
+type taskComparer struct {
+	taskEngine engine.TaskEngine
+}
+
+// NewTaskComparer creates a new taskComparer.
+func NewTaskComparer(taskEngine engine.TaskEngine) *taskComparer {
+	return &taskComparer{
+		taskEngine: taskEngine,
+	}
+}
+
+// sequenceNumberAccessor implements the SequenceNumberAccessor interface defined in ecs-agent module.
+type sequenceNumberAccessor struct {
+	latestSeqNumberTaskManifest *int64
```

Comment:
It might be I am missing some previous discussion in other PRs - why this variable named `latestSeqNumberTaskManifest` but not `latestTaskManifestSeqNumber` and why is it a `*int64` instead of `int64`?

Reply:
> why this variable named latestSeqNumberTaskManifest but not latestTaskManifestSeqNumber

This variable name is how it is currently named in the existing `agent/acs/handler/task_manifest_handler.go` file (see [here](https://github.com/aws/amazon-ecs-agent/blob/dev/agent/acs/handler/task_manifest_handler.go#L43)). This existing naming was just carried over.

> why is it a *int64 instead of int64?
We need to pass by pointer here. If we were to pass by value instead, then we would not be able to change the latestSeqNumberTaskManifest tracked by the Agent (we would only be able to make changes to a copy of it). See [here](https://github.com/aws/amazon-ecs-agent/pull/3840#discussion_r1287767837) for previous similar discussion on this.